src/vmm/src/vstate/memory.rs

```diff
@@ -495,40 +495,12 @@ mod tests {
 
     use super::*;
 
-    fn fork_and_run(function: &dyn Fn(), expect_sigsegv: bool) {
```

Comment:
From what I can understand, forking test harness when it runs all vmm crate tests causes test to hang. Test passes fine if it is run separately or in a small group of tests (like only tests in `memory.rs`) There is another way to test guard pages with treads (by causing separate thread to segfault and panic and than main thread can catch the panic). But this approach does not work with musl target.

Reply:
Ok, I think I understand. Maybe this could be a good target for kani. WDYT @roypat?

Reply:
Mhh, I'm not sure. On the one hand, having kani harnesses for different parts of the firecracker codebase that interact with guest memory will ensure that the guard pages won't be triggered (as the proofs will include that no out of bound guest memory access will happen), but it doesnt exactly show that they work as intended (it only shows that they're not _needed_). But I can't directly see how kani could reason about that, because it doesn't really interact with signals and such.

On the other hand, we could make the test be gnu-only. It might not be out release target, but at least we'd be testing the functionality at all